runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -169,10 +170,13 @@ public Table(@Nonnull final TableParameters<K, V, M> tableParameters,
      * Delete a record mapped to the specified key.
      *
      * @param key Key.
-     * @return Previously stored Corfu Record.
+     * @return Previously stored Corfu Record. Throws NoSuchElementException if key is not found.
      */
     @Nullable
     CorfuRecord<V, M> deleteRecord(@Nonnull final K key) {
+        if (!corfuTable.containsKey(key)) {
+            throw new NoSuchElementException("deleteRecord failed as key "+key.toString()+" does not exist");
+        }
```

Comment:
might be good to remove the unused fields and imports at this time?

Reply:
done, thanks